control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -893,6 +983,43 @@ func createKubeAPIServerService(client client.Client, hcp *hyperv1.HostedControl
 	return svc, nil
 }
 
+func createKubeAPIServerServiceNodePort(client client.Client, hcp *hyperv1.HostedControlPlane, namespace string, status *InfrastructureStatus) (*corev1.Service, error) {
+	svc := &corev1.Service{}
+	svc.Namespace = namespace
+	svc.Name = kubeAPIServerServiceName
+	svc.Spec.Selector = map[string]string{"app": "kube-apiserver"}
+	svc.Spec.Type = corev1.ServiceTypeNodePort
+	var nodePort int32 = 0
+	if hcp.Spec.Services != nil {
+		for _, servicePortSpecification := range hcp.Spec.Services {
+			if servicePortSpecification.Service == hyperv1.APIServer {
+				if servicePortSpecification.ServicePublishingStrategy.NodePort != nil {
+					status.APIAddress = servicePortSpecification.ServicePublishingStrategy.NodePort.Address
+					if servicePortSpecification.ServicePublishingStrategy.NodePort.Port > 0 {
+						nodePort = servicePortSpecification.ServicePublishingStrategy.NodePort.Port
+					}
+				}
+			}
+		}
+	}
+	svc.Spec.Ports = []corev1.ServicePort{
+		{
+			Port:       6443,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(6443),
+		},
+	}
+	if nodePort > 0 {
+		svc.Spec.Ports[0].NodePort = nodePort
+	}
+	svc.OwnerReferences = ensureHCPOwnerRef(hcp, svc.OwnerReferences)
+	err := client.Create(context.TODO(), svc)
```

Comment:
my previous comment should have been here

Reply:
noted! adjusting